src/main/java/net/masterthought/cucumber/util/Util.java

```diff
@@ -76,4 +81,16 @@ public static String toValidFileName(String fileName) {
         // adds MAX_VALUE to eliminate minus character which might be returned by hashCode()
         return Long.toString((long) fileName.hashCode() + Integer.MAX_VALUE);
     }
+
+    /**
+     * Helper method that removes empty hooks from passed array and packs it into new collection.
+     *
+     * @param hooks hooks to be reduced
+     * @return no empty hooks
+     */
+    public static List<Hook> eliminateEmptyHooks(Hook[] hooks) {
```

Comment:
I am not familiar with the template engine. Could the different types cause problems? You get an array, and return a java.util.List object.
In hooks.vm below you _update_ $hooks with the method result, so you change its type. Do the macros support this?

Reply:
it converts smoothly between array and collections